app/src/main/java/meme5/c4q/nyc/meme_project/MainActivity.java

```diff
@@ -1,15 +1,104 @@
 package meme5.c4q.nyc.meme_project;
 
 import android.app.Activity;
-import android.os.Bundle;
 
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+import java.io.File;
+import java.io.FileOutputStream;
 
 public class MainActivity extends Activity {
 
+    private static final int REQUEST_IMAGE_CAPTURE = 1;
+    private Bitmap thumbnail;
+    private String imgFilePath;
+    private String thumbFileName;
+    private boolean imageSelected;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        imageSelected = false;
+        Log.d("onCreate Launched", "onCreate");
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Log.d("onResume Launched", "onResume");
+        if(imageSelected){
+            launchChooseMeme();
+        }
+    }
+
+    private void launchChooseMeme(){
+        Intent chooseMeme = new Intent(this,ChooseMemeStyle.class);
+        chooseMeme.putExtra("thumbnailFileName", thumbFileName);
```

Comment:
The thumbnailFileName is to retrieve the thumbnail file from the apps root private directory. The file name is used to search the directory and find the file to be used later on. If you don't want to pass the name through the intent and just have the file name always be "thumbnail.png", we just all have to know the name of the file if we want to use the thumbnail. We can do that but I believe it's best practice to have a filename variable in case we decide to change the name, we'd only have to do it in one place as opposed to all the places it was used in. @poojawins